pkg/aggregate/iterator.go

```diff
@@ -176,6 +177,11 @@ func (as *AggregateSeries) NewSetFromChunks(length int) *AggregateSet {
 	for _, aggr := range rawAggregators {
 		if aggr&as.aggrMask != 0 {
 			dataArrays[aggr] = make([]float64, length, length) // TODO: len/capacity & reuse (pool)
+			if aggr == aggrTypeMax || aggr == aggrTypeMin || aggr == aggrTypeLast {
+				for i := 0; i < length; i++ {
+					dataArrays[aggr][i] = math.NaN()
```

Comment:
Note, we use to initialise the data arrays with `(+/-)Inf` rather than `NaN`

Reply:
@imakhlin as discussed i think its simpler/cleaner (not confined to expr constraints), u can change if to the other way if u think its critical